### ISSUE TITLE
Test fails on pytest==3.9.1 for py3

### ISSUE THREAD

**Issue by kaisen** [dnephin/PyStaticConfiguration#103]
The test suite `config_test.py::TestConfigurationWatcher` fails with the latest version of pytest (v3.9.1).

```
C02W31HHHTD8:PyStaticConfiguration kaisen$ source .tox/py36/bin/activate
(py36) C02W31HHHTD8:PyStaticConfiguration kaisen$ py.test tests/config_test.py::TestConfigurationWatcher::test_get_filename_list_from_string
=============================================================================== test session starts ===============================================================================
platform darwin -- Python 3.6.5, pytest-3.9.1, py-1.7.0, pluggy-0.8.0
rootdir: /Users/kaisen/github/PyStaticConfiguration, inifile:
collected 1 item

tests/config_test.py
INTERNALERROR> Traceback (most recent call last):
INTERNALERROR>   File "/Users/kaisen/github/PyStaticConfiguration/.tox/py36/lib/python3.6/site-packages/_pytest/main.py", line 181, in wrap_session
INTERNALERROR>     session.exitstatus = doit(config, session) or 0
INTERNALERROR>   File "/Users/kaisen/github/PyStaticConfiguration/.tox/py36/lib/python3.6/site-packages/_pytest/main.py", line 221, in _main
INTERNALERROR>     config.hook.pytest_runtestloop(session=session)
INTERNALERROR>   File "/Users/kaisen/github/PyStaticConfiguration/.tox/py36/lib/python3.6/site-packages/pluggy/hooks.py", line 284, in __call__
INTERNALERROR>     return self._hookexec(self, self.get_hookimpls(), kwargs)
INTERNALERROR>   File "/Users/kaisen/github/PyStaticConfiguration/.tox/py36/lib/python3.6/site-packages/pluggy/manager.py", line 67, in _hookexec
INTERNALERROR>     return self._inner_hookexec(hook, methods, kwargs)
INTERNALERROR>   File "/Users/kaisen/github/PyStaticConfiguration/.tox/py36/lib/python3.6/site-packages/pluggy/manager.py", line 61, in <lambda>
INTERNALERROR>     firstresult=hook.spec.opts.get("firstresult") if hook.spec else False,
INTERNALERROR>   File "/Users/kaisen/github/PyStaticConfiguration/.tox/py36/lib/python3.6/site-packages/pluggy/callers.py", line 208, in _multicall
INTERNALERROR>     return outcome.get_result()
INTERNALERROR>   File "/Users/kaisen/github/PyStaticConfiguration/.tox/py36/lib/python3.6/site-packages/pluggy/callers.py", line 80, in get_result
INTERNALERROR>     raise ex[1].with_traceback(ex[2])
INTERNALERROR>   File "/Users/kaisen/github/PyStaticConfiguration/.tox/py36/lib/python3.6/site-packages/pluggy/callers.py", line 187, in _multicall
INTERNALERROR>     res = hook_impl.function(*args)
INTERNALERROR>   File "/Users/kaisen/github/PyStaticConfiguration/.tox/py36/lib/python3.6/site-packages/_pytest/main.py", line 242, in pytest_runtestloop
INTERNALERROR>     item.config.hook.pytest_runtest_protocol(item=item, nextitem=nextitem)
INTERNALERROR>   File "/Users/kaisen/github/PyStaticConfiguration/.tox/py36/lib/python3.6/site-packages/pluggy/hooks.py", line 284, in __call__
INTERNALERROR>     return self._hookexec(self, self.get_hookimpls(), kwargs)
INTERNALERROR>   File "/Users/kaisen/github/PyStaticConfiguration/.tox/py36/lib/python3.6/site-packages/pluggy/manager.py", line 67, in _hookexec
INTERNALERROR>     return self._inner_hookexec(hook, methods, kwargs)
INTERNALERROR>   File "/Users/kaisen/github/PyStaticConfiguration/.tox/py36/lib/python3.6/site-packages/pluggy/manager.py", line 61, in <lambda>
INTERNALERROR>     firstresult=hook.spec.opts.get("firstresult") if hook.spec else False,
INTERNALERROR>   File "/Users/kaisen/github/PyStaticConfiguration/.tox/py36/lib/python3.6/site-packages/pluggy/callers.py", line 208, in _multicall
INTERNALERROR>     return outcome.get_result()
INTERNALERROR>   File "/Users/kaisen/github/PyStaticConfiguration/.tox/py36/lib/python3.6/site-packages/pluggy/callers.py", line 80, in get_result
INTERNALERROR>     raise ex[1].with_traceback(ex[2])
INTERNALERROR>   File "/Users/kaisen/github/PyStaticConfiguration/.tox/py36/lib/python3.6/site-packages/pluggy/callers.py", line 187, in _multicall
INTERNALERROR>     res = hook_impl.function(*args)
INTERNALERROR>   File "/Users/kaisen/github/PyStaticConfiguration/.tox/py36/lib/python3.6/site-packages/_pytest/runner.py", line 71, in pytest_runtest_protocol
INTERNALERROR>     runtestprotocol(item, nextitem=nextitem)
INTERNALERROR>   File "/Users/kaisen/github/PyStaticConfiguration/.tox/py36/lib/python3.6/site-packages/_pytest/runner.py", line 80, in runtestprotocol
INTERNALERROR>     rep = call_and_report(item, "setup", log)
INTERNALERROR>   File "/Users/kaisen/github/PyStaticConfiguration/.tox/py36/lib/python3.6/site-packages/_pytest/runner.py", line 167, in call_and_report
INTERNALERROR>     hook = item.ihook
INTERNALERROR>   File "/Users/kaisen/github/PyStaticConfiguration/.tox/py36/lib/python3.6/site-packages/_pytest/nodes.py", line 116, in ihook
INTERNALERROR>     return self.session.gethookproxy(self.fspath)
INTERNALERROR>   File "/Users/kaisen/github/PyStaticConfiguration/.tox/py36/lib/python3.6/site-packages/_pytest/main.py", line 422, in gethookproxy
INTERNALERROR>     my_conftestmodules = pm._getconftestmodules(fspath)
INTERNALERROR>   File "/Users/kaisen/github/PyStaticConfiguration/.tox/py36/lib/python3.6/site-packages/_pytest/config/__init__.py", line 383, in _getconftestmodules
INTERNALERROR>     if path.isfile():
INTERNALERROR>   File "/Users/kaisen/github/PyStaticConfiguration/.tox/py36/lib/python3.6/site-packages/py/_path/common.py", line 313, in isfile
INTERNALERROR>     return self.check(file=1)
INTERNALERROR>   File "/Users/kaisen/github/PyStaticConfiguration/.tox/py36/lib/python3.6/site-packages/py/_path/local.py", line 378, in check
INTERNALERROR>     return not kw["file"] ^ isfile(self.strpath)
INTERNALERROR>   File "/Users/kaisen/github/PyStaticConfiguration/.tox/py36/bin/../lib/python3.6/genericpath.py", line 33, in isfile
INTERNALERROR>     return stat.S_ISREG(st.st_mode)
INTERNALERROR> TypeError: an integer is required

========================================================================== no tests ran in 0.05 seconds ===========================================================================
(py36) C02W31HHHTD8:PyStaticConfiguration kaisen$
```

Pinning pytest to v3.8.2 in tox.ini makes the test suite pass. This seems to be caused by how pytest deals with the mocking of `staticconf.config.os.stat` but I haven't been able to narrow it down further.

**Comment by sdelafond** [comment]
I don't see this issue with pytest 3.10 for python3.

**Comment by kaisen** [comment]
Appears to be fixed by #105
